plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ServletClasspathProvider.java

```diff
@@ -25,17 +25,18 @@ public ServletClasspathProvider() {
       return new IClasspathEntry[0];
     } else {
       return resolveClasspathContainer(runtime);
-    } 
+    }
   }
 
   @Override
   public IClasspathEntry[] resolveClasspathContainer(IRuntime runtime) {
-    java.nio.file.Path cloudSdkPath = new CloudSdkProvider(null).getCloudSdkLocation().toPath();
+    java.nio.file.Path cloudSdkPath =
+        new CloudSdkProvider().getCloudSdk().getJavaAppEngineSdkPath();
     if (cloudSdkPath == null) {
       return new IClasspathEntry[0];
     };
-    String servletJar = cloudSdkPath + "/platform/google_appengine/google/appengine/tools/java/lib/shared/servlet-api.jar";
-    String jspJar = cloudSdkPath + "/platform/google_appengine/google/appengine/tools/java/lib/shared/jsp-api.jar";
+    String servletJar = cloudSdkPath + "/shared/servlet-api.jar";
```

Comment:
There are other jars in the location returned by getJavaAppEngineSdkPath().

Changed to use resolve(), though @briandealwis raised a good point: should the library provide methods to get the locations for all needed JARs or should we assume some client knowledge of the SDK structure?

I can't make my mind up on this. I prefer the first, but if there are many JARs the library must discover, the API might get a bit ugly?